packages/wallet-standard/src/wallet.ts

```diff
@@ -0,0 +1,287 @@
+            this.#emit('change', { accounts: this.accounts });
+        }
+    };
+
+    #reconnected = () => {
+        if (window.glowSolana.publicKey) {
+            this.#connected();
+        } else {
+            this.#disconnected();
+        }
+    };
+
+    #connect: ConnectMethod = async ({ silent } = {}) => {
+        await window.glowSolana.connect(silent ? { onlyIfTrusted: true } : undefined);
+
+        this.#connected();
+
+        return { accounts: this.accounts };
+    };
+
+    #disconnect: DisconnectMethod = async () => {
+        await window.glowSolana.disconnect();
+    };
+
+    #signAndSendTransaction: SolanaSignAndSendTransactionMethod = async (...inputs) => {
+        if (!this.#account) throw new Error('not connected');
+
+        const outputs: SolanaSignAndSendTransactionOutput[] = [];
+
+        if (inputs.length === 1) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const { transaction, account, chain, options } = inputs[0]!;
+            const { commitment } = options || {};
+            if (account !== this.#account) throw new Error('invalid account');
+            if (!isSolanaChain(chain)) throw new Error('invalid chain');
+
+            const { signature } = await window.glow.signAndSendTransaction({
+                transactionBase64: Buffer.from(transaction).toString('base64'),
+                network: getNetworkForChain(chain),
+                // HACK: Glow only waits for `confirmed`, so if we want `finalized`, still wait, but don't rely on this.
+                waitForConfirmation: commitment === 'confirmed' || commitment === 'finalized',
+            });
+
+            outputs.push({ signature: decode(signature) });
+        } else if (inputs.length > 1) {
+            for (const input of inputs) {
+                outputs.push(...(await this.#signAndSendTransaction(input)));
+            }
+        }
+
+        return outputs;
+    };
+
+    #signTransaction: SolanaSignTransactionMethod = async (...inputs) => {
+        if (!this.#account) throw new Error('not connected');
+
+        const outputs: SolanaSignTransactionOutput[] = [];
+
+        if (inputs.length === 1) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const { transaction, account, chain } = inputs[0]!;
+            if (account !== this.#account) throw new Error('invalid account');
+            if (chain && !isSolanaChain(chain)) throw new Error('invalid chain');
+
+            const { signedTransactionBase64 } = await window.glow.signTransaction({
+                transactionBase64: Buffer.from(transaction).toString('base64'),
+                // FIXME: Glow's defaults to mainnet, so simulation fails and we can't sign
+                // HACK: Glow's type definition requires `network` but we might not know it, so pass undefined and pray.
+                network: chain ? getNetworkForChain(chain) : (undefined as any as Network),
```

Comment:
I've left these comments in for now since this is a straight copy over from the Wallet Standard repo, but we can remove them if this is desired (per chatting with @vpontis).